pyiron_contrib/generic/coscineIo.py

```diff
@@ -326,24 +332,54 @@ def upload_job(
             form["DOIs"] = dois
 
         self.upload_file(job_file, form)
+        if upload_working_dir:
+            if not job.is_compressed():
+                job.compress()
+            self.upload_file(
+                os.path.join(job.working_directory, job.job_name + ".tar.bz2"), form
```

Comment:
`pyiron_base` has a function that constructs the archive name, best to use it.  (I forgot the name right now, but it's in the same module as the compress/decompress functions)

Reply:
Ah, sorry that was part of https://github.com/pyiron/pyiron_base/pull/969 which I didn't get around to merge.